modin/core/execution/dispatching/factories/factories.py

```diff
@@ -651,3 +651,72 @@ def prepare(cls):
 @doc(_doc_factory_class, execution_name="experimental remote HdkOnNative")
 class ExperimentalHdkOnCloudnativeFactory(ExperimentalRemoteFactory):
     wrapped_factory = ExperimentalHdkOnNativeFactory
+
+
+@doc(_doc_factory_class, execution_name="PandasOnUnidist")
+class PandasOnUnidistFactory(BaseFactory):
+    @classmethod
+    @doc(_doc_factory_prepare_method, io_module_name="``PandasOnUnidistIO``")
+    def prepare(cls):
+        from modin.core.execution.unidist.implementations.pandas_on_unidist.io import (
+            PandasOnUnidistIO,
+        )
+
+        cls.io_cls = PandasOnUnidistIO
+
+
+@doc(_doc_factory_class, execution_name="experimental PandasOnUnidist")
+class ExperimentalPandasOnUnidistFactory(
+    ExperimentalBaseFactory, PandasOnUnidistFactory
+):
```

Comment:
Inheriting from `ExperimentalPandasOnRayFactory` class will help avoid code duplication.

Reply:
Copying an interface that requires another engine to be installed looks wrong, doesn't it?

Reply:
That sounds wrong but still I want unidist to only depend on general stuff but not engines specific.